core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,515 @@
+            }
+          }
+        } catch (EvaluationException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+    return list;
+  }
+
+  public String getTitleBlockCellLabel(EObject diagram, EObject cell) {
+    init();
+    if (cell instanceof DAnnotation) {
+      String name = ((DAnnotation) cell).getDetails().get(NAME);
+      if (name != null)
+        return name;
+    }
+    return "";
+  }
+
+  public String getCellLabel(EObject obj) {
+    if (obj instanceof DAnnotation) {
+      DAnnotation annotation = (DAnnotation) obj;
+      return annotation.getDetails().get(CONTENT);
+    }
+    return "";
+  }
+
+  public boolean isAnnotation(EObject obj) {
+    return obj instanceof DAnnotation;
+  }
+
+  public boolean isTitleBlockAllowed(EObject element, EObject containerView) {
+    if (containerView instanceof DRepresentation) {
+      return true;
+    }
+    return false;
+  }
+
+  public List<DAnnotation> getAvailableTitleBlocksToInsert(final EObject elementView, boolean isDiagramTitleBlock) {
+    List<DAnnotation> result = new ArrayList<>();
+    EList<DAnnotation> eList = ((DDiagram) elementView).getEAnnotations();
+    for (DAnnotation elem : eList) {
+      if (elem.getSource().equals(TITLE_BLOCK)) {
+        result.add(elem);
+      }
+    }
+    List<DAnnotation> finalResult = new ArrayList<>();
+    for (DAnnotation annotation : result) {
+      if (isDiagramTitleBlock) {
+        if (isDiagramTitleBlock(annotation)) {
+          finalResult.add(annotation);
+        }
+      } else {
+        if (!isDiagramTitleBlock(annotation)) {
+          finalResult.add(annotation);
+        }
+      }
+    }
+    return finalResult;
+  }
+
+  public List<DAnnotation> getInitialSelectionOfShowHideTitleBlocks(final EObject elementView,
+      boolean isDiagramTitleBlock) {
+    List<DAnnotation> result = new ArrayList<>(1);
+    DSemanticDiagram diagram = (DSemanticDiagram) CapellaServices.getService().getDiagramContainer(elementView);
+    if (elementView.equals(diagram)) {
+      EList<DAnnotation> diagramElements = diagram.getEAnnotations();
+      for (DAnnotation dDiagramElement : diagramElements) {
+        if (dDiagramElement.getSource().equals(TITLE_BLOCK)) {
+          if (null != dDiagramElement.getDetails().get(VISIBILITY)) {
+            if (dDiagramElement.getDetails().get(VISIBILITY).equals(TRUE)) {
+              result.add(dDiagramElement);
+            }
+          }
+        }
+      }
+    }
+    List<DAnnotation> finalResult = new ArrayList<>();
+    for (DAnnotation annotation : result) {
+      if (isDiagramTitleBlock) {
+        if (isDiagramTitleBlock(annotation)) {
+          finalResult.add(annotation);
+        }
+      } else {
+        if (!isDiagramTitleBlock(annotation)) {
+          finalResult.add(annotation);
+        }
+      }
+    }
+    return finalResult;
+  }
+
+  public EObject showHideTitleBlocks(EObject context, List<DAnnotation> selectedTitleBlocks, DDiagram diagram,

```

Comment:
also here, create one function for diagram TB and one for element TB.

Reply:
Done.